### PR TITLE
Initial work on site and main content components

### PR DESCRIPTION
**Pull request by darwinlopez**
Includes work on rb-site and rb-main

### CODE REVIEW

File: src/rb-main/demo/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+define([
+    '../index.js',
+    './demo-ctrl',
+    './demo-state',
+    '../../rb-site'
+], function (rbMain, demoCtrl, demoState, rbSite) {
+    var demo = angular
+        .module('rb-main-demo', [rbMain.name,
```
Comment (nwhite89) [inline src/rb-main/demo/index.js]: can you split this out please 
```js
[
  rbMain.name,
  rbSite.name
]
```

File: test/unit/rb-main/rb-main.spec.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            });
+
+        describe('rendering', function () {
+
+            it('should render with a "div" tagname', function () {
+                var main = angular.element(template),
+                    element = $compile(main)($scope);
+
+                $scope.$apply();
+                expect(element[0].tagName.toLowerCase()).toEqual('main');
```
Comment (nwhite89) [inline test/unit/rb-main/rb-main.spec.js]: this test doesn't match the "it should" text your expecting a div ?

File: test/unit/rb-main/rb-main.spec.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        it('should convert attributes on rb-main to attributes on the generated wrapper',
+            function () {
+                var main = $compile('<rb-main anyattr any-attr></rb-main>')($scope);
+
+                $scope.$apply();
+                expect(main[0].hasAttribute('anyattr')).toBe(true);
+                expect(main[0].hasAttribute('any-attr')).toBe(true);
+            });
+
+        describe('rendering', function () {
```
Comment (nwhite89) [inline test/unit/rb-main/rb-main.spec.js]: if theres only one test going on in here don't think you need a separate describe?

File: test/unit/rb-site/rb-site.spec.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        it('should convert attributes on rb-site to attributes on the generated wrapper',
+            function () {
+                var site = $compile('<rb-site anyattr any-attr></rb-site>')($scope);
+
+                $scope.$apply();
+                expect(site[0].hasAttribute('anyattr')).toBe(true);
+                expect(site[0].hasAttribute('any-attr')).toBe(true);
+            });
+
+        describe('rendering', function () {
```
Comment (nwhite89) [inline test/unit/rb-site/rb-site.spec.js]: same as previous

File: test/unit/rb-main/rb-main.spec.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+            it('should render with a "div" tagname', function () {
+                var main = angular.element(template),
+                    element = $compile(main)($scope);
+
+                $scope.$apply();
+                expect(element[0].tagName.toLowerCase()).toEqual('main');
+            });
```
Comment (nwhite89) [inline test/unit/rb-main/rb-main.spec.js]: can you please add a test to ensure that content you add within the ``<rb-main>`` is rendered out?

File: test/unit/rb-site/rb-site.spec.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+            it('should render with a "div" tagname', function () {
+                var site = angular.element(template),
+                    element = $compile(site)($scope);
+
+                $scope.$apply();
+                expect(element[0].tagName.toLowerCase()).toEqual('div');
+            });
```
Comment (nwhite89) [inline test/unit/rb-site/rb-site.spec.js]: Could you do the same about rendering content within ``<rb-site>``

File: test/unit/rb-main/rb-main.spec.js (hunk 2 above)
Comment (darwinlopez) [inline test/unit/rb-main/rb-main.spec.js]: Done

File: test/unit/rb-main/rb-main.spec.js (hunk 5 above)
Comment (darwinlopez) [inline test/unit/rb-main/rb-main.spec.js]: Done

File: test/unit/rb-site/rb-site.spec.js (hunk 6 above)
Comment (darwinlopez) [inline test/unit/rb-site/rb-site.spec.js]: Done

File: src/rb-main/rb-main.tpl.html [hunk 7]
```diff
@@ -0,0 +1,2 @@
+<main class="Main Main--background Main--center" role="main" ng-transclude>
+</main>
```
Comment (chrisvxd) [inline src/rb-main/rb-main.tpl.html]: `Main--background Main--center` are both modifiers and shouldn't be on main by default!

**Pull request merged**